crates/matrix-sdk/src/widget/matrix.rs

```diff
@@ -116,14 +121,20 @@ impl MatrixDriver {
         delayed_event_parameters: Option<delayed_events::DelayParameters>,
     ) -> Result<SendEventResponse> {
         let type_str = event_type.to_string();
-        Ok(match (state_key, delayed_event_parameters) {
-            (None, None) => SendEventResponse::from_event_id(
+
+        let redacts =
+            from_raw_json_value::<RedactBody, serde_json::Error>(&content).ok().map(|b| b.redacts);
```

Comment:
Instead of adding `redacts` to the match below, can we do

```rust
if let Some(redacts) = ... {
  return Ok(self.room.redact(&redacts, None, None).await?.event_id);
}
```

Reply:
Yea that also work. I was torn between this which keeps the match expressions small and the readibility by having all possible matrix driver sends in one single match statement.
Thanks for providing you opinion!